lightning/src/ln/channelmanager.rs

```diff
@@ -2090,7 +2090,16 @@ impl<ChanSigner: ChannelKeys, M: Deref> ChannelManager<ChanSigner, M> where M::T
 					if chan.get().get_their_node_id() != *their_node_id {
 						return Err(MsgHandleErrInternal::send_err_msg_no_close("Got a message for a channel from the wrong node!", msg.channel_id));
 					}
-					let monitor_update = try_chan_entry!(self, chan.get_mut().funding_signed(&msg), channel_state, chan);
+					let monitor_update = match chan.get_mut().funding_signed(&msg) {
```

Comment:
14d5b97

This part shouldn't be in previous commit or is there anything new in this one which prevent its?

Reply:
The difference is we're not allowed to lose a monitor update anymore - previously if we were waiting on the user to finish updating a monitor we could just return 				`ChannelError::Ignore("Previous monitor update failure ...")))` as we'd provide them a full copy again later, but now we have to make sure any monitor updates generated are returned even with an Err.